src/coreclr/jit/optimizer.cpp

```diff
@@ -7977,12 +8009,26 @@ bool Compiler::optComputeLoopSideEffectsOfBlock(BasicBlock* blk)
 // Marks the containsCall information to "lnum" and any parent loops.
 void Compiler::AddContainsCallAllContainingLoops(unsigned lnum)
 {
+    unsigned nestedLoopNum = lnum;
     assert(0 <= lnum && lnum < optLoopCount);
     while (lnum != BasicBlock::NOT_IN_LOOP)
     {
         optLoopTable[lnum].lpContainsCall = true;
         lnum                              = optLoopTable[lnum].lpParent;
     }
+
+#ifdef FEATURE_LOOP_ALIGN
```

Comment:
I don't remember why I don't do this right away instead of storing `nestedLoopNum `. I will move this.

Reply:
You mean check this when you set the align flag above? Makes sense instead of an "undo" here.

Reply:
> You mean check this when you set the align flag above?

Meaning, just remove the flag as the first thing when we enter in this method instead of doing it after the `while` loop.

>  Makes sense instead of an "undo" here.

I do not completely follow it. I am assuming you agree to what I am thinking.

Reply:
My comment wasn't clear.

Why not defer marking aligned loops until after we've computed if they contain calls, rather than marking them and then unmarking them.....?

Reply:
Sure. In that case, we miss out marking certain loops as need alignment specially the ones that get cloned. Having `BBF_LOOP_ALIGN` initially when we find natural loops help us propagate the flag to the cloned loop as well. Today, `optLoopTable` do not track cloned loop (probably because we know it is a slow path? @BruceForstall ) . So after we are done with value numbering (where we mark if the loop has call or not), and if we just rely on `optLoopTable` to mark the alignment flag, we would miss the ones that got cloned before value numbering.